src/UseCases/ModerateMembershipApplication/ModerateMembershipApplicationUseCase.php

```diff
@@ -27,15 +29,16 @@ public function markMembershipApplicationAsModerated( int $membershipApplication
 			return ModerateMembershipApplicationResponse::newFailureResponse( $membershipApplicationId );
 		}
 
-		if ( $membershipApplication->needsModeration() ) {
+		// TODO should this be thrown out?
+		if ( $membershipApplication->isMarkedForModeration() ) {
```

Comment:
It doesn't matter what moderation reasons exist (flagged by admin or auto-moderated) - an admin should only be able to remove the moderations, not add new ones. In the FOC the moderation actions should stay "binary" - remove (all moderations, regardless of type) and add (with reason "flagged by admin"). This use case is for user actions that change the moderation reasons, so there should be a check against "double-moderation".